Sources/SwiftFormatRules/DeclSyntax+Comments.swift

```diff
@@ -0,0 +1,145 @@
+        var isTheFirstLine = true
+        let blockCommentWithoutMarks = blockComment.map { (line: String) -> String in
+          // Only the first line of the block comment start with '/**'
+          let markToRemove = isTheFirstLine ? "/**" : "* "
+          let trimmedLine = line.trimmingCharacters(in: .whitespaces)
+          if trimmedLine.starts(with: markToRemove) {
+            let numCharsToRemove = isTheFirstLine ? markToRemove.count : markToRemove.count - 1
+            isTheFirstLine = false
+            return trimmedLine.hasSuffix("*/")
+              ? String(trimmedLine.dropFirst(numCharsToRemove).dropLast(3)) : String(
+                trimmedLine.dropFirst(numCharsToRemove))
+          } else if trimmedLine == "*" {
+            return ""
+          } else if trimmedLine.hasSuffix("*/") {
+            return String(line.dropLast(3))
+          }
+          isTheFirstLine = false
+          return line
+        }
+
+        return blockCommentWithoutMarks.joined(separator: "\n").trimmingCharacters(in: .newlines)
+      case .docLineComment(let text):
+        // Mark that we've started grabbing sequential line comments and append it to the
+        // comment buffer.
+        hasSeenFirstLineComment = true
+        comment.append(text)
+      case .newlines(let n), .carriageReturns(let n), .carriageReturnLineFeeds(let n):
+        // Only allow for 1 newline between doc line comments, but allow for newlines between the
+        // doc comment and the declaration.
+        guard n == 1 || !hasSeenFirstLineComment else { break gatherComments }
+      case .spaces, .tabs:
+        // Skip all spaces/tabs. They're irrelevant here.
+        break
+      default:
+        if hasSeenFirstLineComment {
+          break gatherComments
+        }
+      }
+    }
+
+    /// Removes the "///" from every line of comment
+    let docLineComments = comment.reversed().map { $0.dropFirst(3) }
+    return comment.isEmpty ? nil : docLineComments.joined(separator: "\n")
+  }
+
+  var docCommentInfo: ParseComment? {
+    guard let docComment = self.docComment else { return nil }
+    let comments = docComment.components(separatedBy: .newlines)
+    var params = [ParseComment.Parameter]()
+    var commentParagraphs = [String]()
+    var hasFoundParameterList = false
+    var hasFoundReturn = false
+    var returnsDescription: String?
+    // Takes the first sentence of the comment, and counts the number of lines it uses.
+    let oneSenteceSummary = docComment.components(separatedBy: ".").first
+    let numOfOneSentenceLines = oneSenteceSummary!.components(separatedBy: .newlines).count
+
+    // Iterates to all the comments after the one sentence summary to find the parameter(s)
+    // return tags and get their description.
+    for line in comments.dropFirst(numOfOneSentenceLines) {
+      let trimmedLine = line.trimmingCharacters(in: .whitespaces)
+
+      if trimmedLine.starts(with: "- Parameters") {
```

Comment:
We're planning to use cmark to parse doc comments (and potentially reflow them to fit line limits, since cmark can also re-render them, but that will require some changes to cmark itself to preserve the original markup). I don't know if it would make sense to move that into swift-syntax eventually, though, or to expose it in _InternalSyntaxParser.